vaadin-checkbox-flow-parent/vaadin-checkbox-flow/src/main/java/com/vaadin/flow/component/checkbox/FieldValidationUtil.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2000-2022 Vaadin Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.vaadin.flow.component.checkbox;
+
+class FieldValidationUtil {
+    private FieldValidationUtil() {
+    }
+
+    static <T> void disableClientValidation(CheckboxGroup<T> component) {
+        // Since this method should be called for every time when the component
+        // is attached to the UI, lets check that it is actually so
+        if (!component.isAttached()) {
+            throw new IllegalStateException(String.format(
+                    "Component %s is not attached. Client side validation "
+                            + "should be disabled when the component is "
+                            + "attached, thus this method needs to be called "
+                            + "from the onAttach method of the component.",
+                    component));
+        }
+        // Wait until the response is being written as the validation state
+        // should not change after that
+        component.getUI().ifPresent(ui -> ui.beforeClientResponse(component,
+                executionContext -> overrideClientValidation(component)));
+    }
+
+    private static <T> void overrideClientValidation(
+            CheckboxGroup<T> component) {
+        StringBuilder expression = new StringBuilder(
+                "this.validate = function () {return !(this.required && this.value.length === 0);};");
```

Comment:
Had to duplicate the actual validation logic here, as the web component does not have a `checkValidity` method in this version. I'm not sure if the implementation of this method matters at all, since it doesn't really play a role in the server-side validation.